lib/reek/smells/smell_warning.rb

```diff
@@ -53,12 +55,17 @@ def matches_smell_type?(klass)
         smell_classes.include?(klass.to_s)
       end
 
-      def matches_smell_details?(other_smell_details = {})
-        common_parameters_equal?(other_smell_details)
+      def matches_attributes?(other_attributes = {})
+        check_attributes_comparability(other_attributes)
+        return false unless common_parameters_equal?(other_attributes[:parameters])
+        other_attributes.delete(:parameters)
```

Comment:
I'm with @tansaku on this one. I believe our current structure is not quite there but we're on a good way.
I'd rather see this PR finished and merged and then address this in a follow-up.

>> and others that are collected into a open ended group called "parameters"

Exactly. I believe a cleaner, more OOP solution would be to have a class called SmellDetails and then rename the "parameters" attribute above "details" and point that to an instance. But let's leave that for a separate discussion.